### PR TITLE
Update ooc.dm. Adds two new OOC verbs for team-based OOC.

### PR DESCRIPTION
**Pull request by MiniMeatwad**
## About The Pull Request

Copies prefs and spans and bans from OOC because I don't know how to touch those, and since it involves preferences and possibly database stuff: I don't think we modify those until changes are fully merged permanently.

I'm not entirely sure about the safety of the admin-related checks since I didn't look into all of the admin code, but when I tested in-game alone (I don't know how to get multiple clients on a local server), it was fine.

## Why It's Good For The Game

So people can talk to their entire team through OOC means.

## Changelog
:cl:
add: Adds Team OOC, two new verbs in the OOC tab called XOOC and MOOC for Xeno and Marine.
code: Modified ooc.dm, adding two new verbs.
admin: XOOC and MOOC should have clickable player panel links for admins, and admins can see XOOC and MOOC chat and talk in them.
/:cl:

Screenshot, 2 sequential tests for admin and non-admin, and for observer/human/xeno: For debug, verbs were made callable with no checks. Observer Team OOC as non-admin is not supposed to happen, so your key wont be revealed if it does happen.
dreamseeker_2XPuBEcgtn

### CODE REVIEW

**Comment by Alexeivichi** [comment]
Been hoping for something like. You can connect multiple clients to one local server on the same machine by just logging in with a different user on your Byond pager, and connecting via Dream Maker as normal.

**Comment by MiniMeatwad** [comment]
> 
> 
> Been hoping for something like. You can connect multiple clients to one local server on the same machine by just logging in with a different user on your Byond pager, and connecting via Dream Maker as normal.

Thanks.

File: code/game/verbs/ooc.dm [hunk 1]
```diff
@@ -107,6 +107,181 @@
...
+	if(!mob)
+		return
+	if(IsGuestKey(key))
+		to_chat(src, "Guests may not use XOOC.")
+		return
+	if(mob.stat == DEAD && !admin)
+		to_chat(src, span_warning("You must be alive to use XOOC."))
+		return
+	if(!(mob in GLOB.xeno_mob_list) && !admin)
+		to_chat(src, span_warning("You must be a human to use XOOC."))
```
Comment (hyper2snyper) [inline code/game/verbs/ooc.dm:131]:
```suggestion
		to_chat(src, span_warning("You must be an xeno to use XOOC."))
```

Comment (MiniMeatwad) [reply code/game/verbs/ooc.dm:131]: lol thanks

File: code/game/verbs/ooc.dm [hunk 2]
```diff
@@ -107,6 +107,181 @@
...
+			message_admins("[ADMIN_TPMONTY(usr)] has attempted to advertise in OOC: [msg]")
+			return
+
+	if(is_banned_from(ckey, "OOC"))
+		to_chat(src, span_warning("You have been banned from OOC."))
+		return
+
+	mob.log_talk(msg, LOG_OOC)
+
+	for(var/client/C in GLOB.clients)
```
Comment (BraveMole) [inline code/game/verbs/ooc.dm:254]:
```suggestion
	for(var/client/C AS in GLOB.clients)
```

File: code/game/verbs/ooc.dm [hunk 3]
```diff
@@ -107,6 +107,181 @@
...
+		return
+
+	msg = emoji_parse(msg)
+
+	if(!(prefs.toggles_chat & CHAT_OOC))
+		to_chat(src, span_warning("You have OOC muted."))
+		return
+
+	if(!check_rights(R_ADMIN, FALSE))
+		if(!GLOB.ooc_allowed)
```
Comment (BraveMole) [inline code/game/verbs/ooc.dm:234]: a way to selectively mute each faction OOC would be nice IMO. Don't remove that though

File: code/game/verbs/ooc.dm [hunk 4]
```diff
@@ -107,6 +107,181 @@
...
+		if(admin && !(mob in GLOB.human_mob_list))
+			display_name = mob.key
+
+		// Admins open straight to player panel
+		if(check_other_rights(C, R_ADMIN, FALSE))
+			display_name = "<a class='hidelink' href='?_src_=holder;[HrefToken(TRUE)];playerpanel=[REF(usr)]'>[display_name]</a>"
+		var/avoid_highlight = C == src
+		to_chat(C, "<font color='#665544'>[span_ooc("<span class='prefix'>MOOC: [display_name]")]: <span class='message linkify'>[msg]</span></span></font>", avoid_highlighting = avoid_highlight)
+
+	for(var/client/C in GLOB.admins)
```
Comment (BraveMole) [inline code/game/verbs/ooc.dm:270]:
```suggestion
	for(var/client/C AS in GLOB.admins)
```

File: code/game/verbs/ooc.dm [hunk 5]
```diff
@@ -107,6 +107,181 @@
...
+		if(admin && !(mob in GLOB.xeno_mob_list))
+			display_name = mob.key
+
+		// Admins open straight to player panel
+		if(check_other_rights(C, R_ADMIN, FALSE))
+			display_name = "<a class='hidelink' href='?_src_=holder;[HrefToken(TRUE)];playerpanel=[REF(usr)]'>[display_name]</a>"
+		var/avoid_highlight = C == src
+		to_chat(C, "<font color='#665544'>[span_ooc("<span class='prefix'>XOOC: [display_name]")]: <span class='message linkify'>[msg]</span></span></font>", avoid_highlighting = avoid_highlight)
+
+	for(var/client/C in GLOB.admins)
```
Comment (BraveMole) [inline code/game/verbs/ooc.dm:184]:
```suggestion
	for(var/client/C AS in GLOB.admins)
```

File: code/game/verbs/ooc.dm [hunk 6]
```diff
@@ -107,6 +107,181 @@
...
+			message_admins("[ADMIN_TPMONTY(usr)] has attempted to advertise in OOC: [msg]")
+			return
+
+	if(is_banned_from(ckey, "OOC"))
+		to_chat(src, span_warning("You have been banned from OOC."))
+		return
+
+	mob.log_talk(msg, LOG_OOC)
+
+	for(var/client/C in GLOB.clients)
```
Comment (BraveMole) [inline code/game/verbs/ooc.dm:168]:
```suggestion
	for(var/client/C AS in GLOB.clients)
```

**Review by BraveMole — CHANGES_REQUESTED** [review]
Pending headmins approval

**Comment by bobmancer343** [comment]
Yess, I have been asking for this concept for a bit. It's good for new players that want to ask for help from fellow marines, xenos.

File: code/game/verbs/ooc.dm (hunk 6 above)
Comment (MiniMeatwad) [reply code/game/verbs/ooc.dm]: What does "AS in" do?

File: code/game/verbs/ooc.dm (hunk 4 above)
Comment (MiniMeatwad) [reply code/game/verbs/ooc.dm]: Changed to "client/C AS in GLOB."

File: code/game/verbs/ooc.dm (hunk 5 above)
Comment (MiniMeatwad) [reply code/game/verbs/ooc.dm]: Changed to "client/C AS in GLOB."

File: code/game/verbs/ooc.dm (hunk 6 above)
Comment (MiniMeatwad) [reply code/game/verbs/ooc.dm]: Changed to "client/C AS in GLOB."

File: code/game/verbs/ooc.dm (hunk 2 above)
Comment (MiniMeatwad) [reply code/game/verbs/ooc.dm]: Changed to "client/C AS in GLOB."

**Comment by lKiseki** [comment]
> Pending headmins approval

Lewd seal of approval (TM).

**Review by BraveMole — CHANGES_REQUESTED** [review]
I don't see any keybinds? Defintly add thos

**Comment by MiniMeatwad** [comment]
> 
> 
> I don't see any keybinds? Defintly add thos

Added, no assigned keys since the keybindings are already cluttered. Users can set them if they need them.

**Review by BraveMole — APPROVED** [review]

**Comment by MiniMeatwad** [comment]
Changed title since someone said there should be less "update file.dm"s in my PRs, now missing TM label

**Review by BraveMole — APPROVED** [review]

**Pull request merged**